tables/data/data.c

```
@@ -127,16 +129,35 @@ void data_write_data_segment(struct DataTable* data, FILE* fout) {
 	data_resize(data);
 
 	struct DataEntry* entry = malloc(sizeof(struct DataEntry));
 
+	if (entry == NULL) {
+		return false;
+	}
+
 	entry->value = strdup(str);
+
+	if (entry->value == NULL) {
+		return false;
```

Cppcheck (2.10-2)

[error] Memory leak

(memleak)